src/components/application_manager/include/application_manager/help_prompt_manager_impl.h

```diff
@@ -0,0 +1,184 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HELP_PROMPT_MANAGER_IMPL_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HELP_PROMPT_MANAGER_IMPL_H_
+
+#include "application_manager/help_prompt_manager.h"
+
+#include <algorithm>
+#include <vector>
+
+#include "utils/lock.h"
+#include "utils/timer.h"
+
+namespace application_manager {
+
+class Application;
+class ApplicationManager;
+
+/**
+ * @brief The HelpPromptManagerImpl class implemented interface
+ * HelpPromptManager
+ * is a mechanism for handling "help" requests
+ * by the user when the application does not provide any or such information.
+ */
+class HelpPromptManagerImpl : public HelpPromptManager {
+ public:
+  /**
+   * @brief Container for buffering VR help commands
+  */
+  typedef std::pair<uint32_t, smart_objects::SmartObjectSPtr> VRCommandPair;
+  typedef std::vector<VRCommandPair> VRCommandPairs;
+
+  /**
+   * @brief Class constructor
+   * @param app owner for this class
+   */
+  HelpPromptManagerImpl(Application& app, ApplicationManager& app_manager);
+
+  /**
+   * @brief Class destructor
+   */
+  ~HelpPromptManagerImpl();
+
+  /**
+   * @brief Adds command to constructed values, and sends SetGlobalProperties if
+   * required
+   * @param cmd_id command unique ID
+   * @param command smart object with commands to add
+   * @param is_resumption flag for identifying if command was added during data
+   * resumption process
+   */
+  void OnVrCommandAdded(const uint32_t cmd_id,
+                        const smart_objects::SmartObject& command,
+                        const bool is_resumption) OVERRIDE;
+
+  /**
+   * @brief Removes command from constructed values, and send
+   * SetGlobalProperties if required
+   * @param cmd_id command unique ID
+   * @param is_resumption flag for identifying if command was added during data
+   * resumption process
+   */
+  void OnVrCommandDeleted(const uint32_t cmd_id,
+                          const bool is_resumption) OVERRIDE;
+
+  /**
+     * @brief Stop constructing vrHelp and/or helpPrompt if they are present in
+     * message
+     * @param msg containing GlobalProperties
+     * @param is_response determines is the request or response for the for
+     * the SetGlobalPropertiesRequest
+     */
+  void OnSetGlobalPropertiesReceived(const smart_objects::SmartObject& msg,
+                                     const bool is_response) OVERRIDE;
+
+  /**
+   * @brief Get current sending type
+   * @return current sending type
+   */
+  SendingType GetSendingType() const OVERRIDE;
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(HelpPromptManagerImpl);
+
+  /**
+   * @brief Add new smart object with VR command to the map
+   * @param cmd_id ID of VR command
+   * @param command smart object containing VR command structure
+   * @return true if command was added successfully otherwise returns false
```

Comment:
Seems that the purpose of the return value has changed, since false is returned if the command limit already exceeds 30, but the command is still added to the list.